### PR TITLE
CBA Settings implementation

### PR DESCRIPTION
**Pull request by Wyqer**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| Needs wipe? | no |
| Fixed issues | #514 |

### Description:
All mission parameters are deleted, as the mission configuration is now implemented via the CBA Settings system. Due to this change, the `wipe` functionality is moved to the server admin dialog. As it's kind of possible that people could click the "wipe save" button by mistake or to "try it", the current save data will be exported to the players profile, before the server wipes all his save data. This way the player can just import the save data again to undo the save wipe. A server/mission restart has to be done in both cases.

The big benefit by using CBA Settings is the possible modularity. Every module can now add their own CBA settings without the need to touch any of the existing files. Furthermore the whole "persistent parameters" functionality is now obsolete, as it's part of the CBA Settings system.

Furthermore I've completely cleaned and recategorized the stringtable.xml. I've also added a tag to the strings, as we didn't had one since now. This is to ensure there won't be any issues with mods etc. which are also using an own stringtable.

All modules are now also contain a readme.md file with an overview about the module functionality and a list of all included functions.

### Content:
- [x] Removed mission parameters
- [x] Cleaned the KPLIB_config.sqf
- [x] Created a backup of the old config file
- [x] Implemented all currently used/needed settings as CBA settings
- [x] Sorted all CBA settings to corresponding module settings (currently just init and respawn)
- [x] Added strings for the CBA settings to the stringtable
- [x] Cleaned and resorted the whole stringtable and added a tag
- [x] Adjusted all files concerning the new settings variables/strings
- [x] Added module description readmes to each module root folder
- [x] Moved save wipe from mission parameters to the admin menu
- [x] Changelog update

### Successfully tested on:
- [x] Local MP Vanilla
- [x] Dedicated MP Vanilla

### Compatibility checked with:
* Wasn't necessary

### CODE REVIEW

File: Missionframework/KPLIB_config_old.sqf [hunk 1]
```diff
@@ -0,0 +1,292 @@
+/*
```
Comment (veteran29) [inline Missionframework/KPLIB_config_old.sqf]: Any good reason to keeps this file? We can always check how it was via git history.

File: Missionframework/modules/00_init/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+The init module handles the general mission initialization and distributes it to all clients.
+It contains the general/vital CBA settings for the mission, defines basic global variables and ui classes/values.
+Also the presets are loaded and checked for syntax errors. If any errors are found, it'll end the mission with a hint.
+Furthermore it takes care of the saving process by running scheduled savings.
+Mostly everything is processed on the server, which distributes the final data to the clients.
+
+### Dependencies
+This module uses no data or functions from other modules.
+
+### Functions
```
Comment (veteran29) [inline Missionframework/modules/00_init/README.md]: The module and dependencies description is an great addition,  I'm not sure if functions needs to be described here. It adds another place where you need to document the function and it basically contains same stuff as `functions.hpp` comments.

File: Missionframework/modules/00_init/fnc/fn_init_settings.sqf [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+// Default: 150 meters
+[
+    "KPLIB_param_sectorCapRange",
+    "SLIDER",
+    [localize "STR_KPLIB_SETTINGS_GENERAL_SECRANGE", localize "STR_KPLIB_SETTINGS_GENERAL_SECRANGE_TT"],
+    localize "STR_KPLIB_SETTINGS_GENERAL",
+    [100, 200, 150, 0],
+    1,
+    {}
+] call CBA_Settings_fnc_init;
```
Comment (veteran29) [inline Missionframework/modules/00_init/fnc/fn_init_settings.sqf]: Sector handling is in core, shouldn't these settings be in core too then?

File: Missionframework/modules/00_init/fnc/fn_init_settings.sqf [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+        ],
+        [
+            localize "STR_A3_None",
+            localize "STR_A3_Medic"
+        ],
+        1
+    ],
+    1,
+    {}
+] call CBA_Settings_fnc_init;
```
Comment (veteran29) [inline Missionframework/modules/00_init/fnc/fn_init_settings.sqf]: BIS revive is also initialized in core.

File: Missionframework/modules/02_core/fnc/fn_core_intro.sqf [hunk 5]
```diff
@@ -21,31 +21,33 @@
 --- TODO IN SPRINT .... soon ^^ ---
 Rework the intro and cinematic. Currently it doesn't work reliable. Mostly an endless loop of the cinematic.
 Postponed, as it's not in the current scope of the sprint 5 tasks.
+
+Also: Refactor or even remove it completely
```
Comment (veteran29) [inline Missionframework/modules/02_core/fnc/fn_core_intro.sqf]: Any reason to keep dead code? Same concern as config, we can check old code via git.

File: Missionframework/KPLIB_config_old.sqf (hunk 1 above)
Comment (Wyqer) [reply Missionframework/KPLIB_config_old.sqf]: Of course, we can always have a look at it via the history. Will be deleted. 👍

File: Missionframework/modules/00_init/README.md (hunk 2 above)
Comment (Wyqer) [reply Missionframework/modules/00_init/README.md]: I added the functions "duplication" there with a wiki page generating process in mind, which is able to read these markdown files. But the site/wiki is your baby of course, so just tell me if we need/keep it or if I should remove it. 🙂

File: Missionframework/modules/00_init/fnc/fn_init_settings.sqf (hunk 3 above)
Comment (Wyqer) [reply Missionframework/modules/00_init/fnc/fn_init_settings.sqf]: I've also thought about that, but came to the opinion that init/common/core are all able to be categorized as "general settings" and therefore I kept this category together in one file.

File: Missionframework/modules/00_init/fnc/fn_init_settings.sqf (hunk 4 above)
Comment (Wyqer) [reply Missionframework/modules/00_init/fnc/fn_init_settings.sqf]: That's true, of course. The BI Revive category will be moved to core then. 👍

File: Missionframework/modules/02_core/fnc/fn_core_intro.sqf (hunk 5 above)
Comment (Wyqer) [reply Missionframework/modules/02_core/fnc/fn_core_intro.sqf]: I wanted to have a discussion in the next days in Discord concerning keeping the intro cinematic or not. Until we come to a decision I haven't really cared much about this particular code/function.

File: Missionframework/modules/00_init/fnc/fn_init_loadPresets.sqf [hunk 6]
```diff
@@ -26,7 +26,7 @@
 */
 
 // Load Blufor preset
-switch (KPLIB_preset_blufor) do {
+switch (KPLIB_param_presetPlayer) do {
```
Comment (Dubjunk) [inline Missionframework/modules/00_init/fnc/fn_init_loadPresets.sqf]: Maybe this param could be named `KPLIB_param_presetFriendly` as it effects the whole faction with all AI's and not only the players.
Could be missunderstood.

File: Missionframework/modules/00_init/fnc/fn_init_settings.sqf [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+            localize "STR_KPLIB_SETTINGS_PRESET_ARSENAL_1",
+            localize "STR_KPLIB_SETTINGS_PRESET_ARSENAL_2"
+        ],
+        1
+    ],
+    1,
+    {}
+] call CBA_Settings_fnc_init;
+
+// KPLIB_param_presetPlayer
```
Comment (Dubjunk) [inline Missionframework/modules/00_init/fnc/fn_init_settings.sqf]: Same as mentioned above.

File: Missionframework/modules/00_init/fnc/fn_init_settings.sqf (hunk 3 above)
Comment (veteran29) [reply Missionframework/modules/00_init/fnc/fn_init_settings.sqf]: I'm not sure. If in the future you will want to edit something related to these settings, will you remember that despite that the code for sectors is in `core` the settings related to it are in `init` module?

File: Missionframework/modules/00_init/README.md (hunk 2 above)
Comment (veteran29) [reply Missionframework/modules/00_init/README.md]: Lets leave it for now.

**Comment by veteran29** [comment]
lgtm

**Pull request merged**